### PR TITLE
Refactored from war-project to jar-project

### PR DESCRIPTION
**Pull request by janvanmansum**
#### When applied it will
- Not change the interface of the `easy-pid-generator`
- Internally the project is now a `jar`-project
#### How should this be manually tested?
- Deploy on deasy an verify that you can still create PIDs the usual way: `curl -X POST http://localhost:8082/pids?type=urn`
- Verify that the service can still be started and stopped as usual: `service easy-pid-generator start|stop|status`

### CODE REVIEW

**Comment by janvanmansum** [comment]
@rvanheest : please, have a look at this (and the related pull request). `easy-pid-generator` can now be started in one of two modes: `rest` or `hazelcast`. The mode is set in `application.conf`. I I have tested teh `rest`-mode and it works. The `hazelcast`-mode works up to the point the the client seems to be able to connect and stay connected. After that it is up to you to make the magic work :)

File: src/main/assembly/dist/cfg/application.conf [hunk 1]
```diff
@@ -1,12 +1,14 @@
+port = "{{ easy_pid_generator_port }}"
+mode = "{{ easy_pid_generator_mode }}"
 types {
    urn {
        namespace = "{{ easy_pid_generator_urn_namespace }}"
        dashPosition = 4
-       firstSeed = {{ easy_pid_generator_urn_first_seed }}
+       firstSeed = "{{ easy_pid_generator_urn_first_seed }}"
```
Comment (rvanheest) [inline src/main/assembly/dist/cfg/application.conf:7]: I'm not sure about these quotes here, as we expect a `Long`...

File: src/main/assembly/dist/cfg/application.conf [hunk 2]
```diff
@@ -1,12 +1,14 @@
+port = "{{ easy_pid_generator_port }}"
```
Comment (rvanheest) [inline src/main/assembly/dist/cfg/application.conf:1]: Also I'm not sure about the `application.conf` style configuration file here. I thought we were migrating to the more simpler `application.properties` format? Should we also do that here or do that in another PR? In any case, I got a TODO in this in my own branch...

File: src/main/scala/nl/knaw/dans/easy/pid/ServiceStarter.scala [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+
+import java.io.File
+
+import com.typesafe.config.ConfigFactory
+import nl.knaw.dans.easy.pid.microservice.HazelcastService
+import org.apache.commons.daemon.{Daemon, DaemonContext}
+import org.eclipse.jetty.server.Server
+import org.slf4j.LoggerFactory
+
+class ServiceStarter extends Daemon {
```
Comment (rvanheest) [inline src/main/scala/nl/knaw/dans/easy/pid/ServiceStarter.scala:26]: I got this in my branch as well, in a slightly different way but it looks quite the same! Accepted for now, I'll merge it with my branch and see if I need to add some things...

File: src/main/scala/nl/knaw/dans/easy/pid/microservice/HazelcastService.scala [hunk 4]
```diff
@@ -19,35 +19,25 @@ import com.hazelcast.Scala.client._
...
     val conf = new ClientConfig()
     serialization.Defaults.register(conf.getSerializationConfig)
     hz = conf.newClient()
-
     PidGeneratorService.run(hz) // can't pass this implicitly since `hz` is a variable
   }
 
-  def stop(): Unit = {
-    log.info("Stopping pid-generator service ...")
+  override def stop(): Unit = {
```
Comment (rvanheest) [inline src/main/scala/nl/knaw/dans/easy/pid/microservice/HazelcastService.scala:38]: **A quick word of preference/style here, which is to be taken as a personal note!**

you don't really need the `override` keyword here, as you are _implementing_ this function here rather than _overriding_ it. To me the `override` means literally _override the implementation_ rather than _implement it for the first time_. Got me confused for a moment :-) On the other hand, it makes it more clear that this method is not original here, but that it is enforced by some other interface/trait/(abstract) class. So there is to say something for both sides!

File: src/main/scala/nl/knaw/dans/easy/pid/microservice/HazelcastService.scala [hunk 5]
```diff
@@ -19,35 +19,25 @@ import com.hazelcast.Scala.client._
 import com.hazelcast.Scala.serialization
 import com.hazelcast.client.config.ClientConfig
 import com.hazelcast.core.HazelcastInstance
+import nl.knaw.dans.easy.pid.Service
 import org.apache.commons.daemon.{Daemon, DaemonContext}
 import org.slf4j.LoggerFactory
 
-class ServiceStarter extends Daemon {
-
+case class HazelcastService(conf: com.typesafe.config.Config) extends Service {
```
Comment (rvanheest) [inline src/main/scala/nl/knaw/dans/easy/pid/microservice/HazelcastService.scala:26]: maybe do an `import com.typesafe.config.Config`?

**Review by rvanheest — COMMENTED** [review]
some comments on this PR. I think we should merge this rather soon, as I require this code for the next step in my code...
